### DISCUSSION TITLE
How relevant is replacing Array.splice today?

### DISCUSSION THREAD

**Discussion by nz771202** [pixijs/pixijs#9460] [General]
In 2015 this was probably perfectly valid as browsers weren't as optimized.
https://github.com/pixijs/pixijs/pull/2209

I'm not expert at benchmarking or memory profiling, but based on my tests Array.splice seems faster now, especially on mobile, and the GC overhead doesn't seem to be an issue. Any thoughts?
https://jsperf.app/hupacu

I think this is kind of important because Container.removeChild is using the replacement function as well, and could probably be made faster with the native Array.splice.

**Comment by miltoncandelero** [comment]
There is more than just raw speed, you have to check what happens to memory in both cases. The garbage collector is a silent killer
